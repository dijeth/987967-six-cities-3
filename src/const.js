const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export {OfferType, CITIES};
