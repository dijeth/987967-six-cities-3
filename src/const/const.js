"use strict";
exports.__esModule = true;
exports.OfferType = {
    APARTMENT: "Apartment",
    PRIVATE_ROOM: "Private room"
};
exports.CITIES = [
    "Paris",
    "Cologne",
    "Brussels",
    "Amsterdam",
    "Hamburg",
    "Dusseldorf"
];
exports.InsideFeature = {
    WI_FI: "Wi-Fi",
    WASHING_MACHINE: "Washing machine",
    TOWELS: "Towels",
    HEATING: "Heating",
    COFFEE_MACHINE: "Coffee machine",
    BABY_SEAT: "Baby seat",
    KITCHEN: "Kitchen",
    DISHWASHER: "Dishwasher",
    CABEL_TV: "Cabel TV",
    FRIDGE: "Fridge"
};
exports.ScreenType = {
    MAIN: "MAIN",
    PROPERTY: "PROPERTY"
};
exports.BREAK_STRING = "\n";
exports.MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
exports.MAX_CITY_COUNT = 6;
var SortType;
(function (SortType) {
    SortType["POPULAR"] = "Popular";
    SortType["PRICE_LOW_TO_HIGH"] = "Price: low to high";
    SortType["PRICE_HIGH_TO_LOW"] = "Price: high to low";
    SortType["TOP_RATED_FIRST"] = "Top rated first";
})(SortType = exports.SortType || (exports.SortType = {}));
exports.SORT_LIST = [
    SortType.POPULAR,
    SortType.PRICE_LOW_TO_HIGH,
    SortType.PRICE_HIGH_TO_LOW,
    SortType.TOP_RATED_FIRST
];
exports.AuthorizationStatus = {
    NO_AUTH: "NO_AUTH",
    AUTH: "AUTH"
};
exports.AppRoute = {
    getLogin: function () { return "/login"; },
    getOffer: function (id) { return "/offer/" + id; },
    getFavorites: function () { return "/favorites"; },
    getRoot: function () { return "/"; }
};
exports.ServerRoute = {
    getLogin: function () { return "/login"; },
    getFavorites: function () { return "/favorite"; },
    postFavorites: function (id, status) { return "/favorite/" + id + "/" + status; },
    getNearby: function (id) { return "/hotels/" + id + "/nearby"; },
    getHotels: function () { return "/hotels"; },
    getComments: function (id) { return "/comments/" + id; }
};
exports.MAX_IMAGE_COUNT = 6;
exports.ReviewLength = {
    MIN: 50,
    MAX: 300
};
exports.ServerError = {
    UNAUTHORIZED: 401,
    BAD_REQUEST: 400,
    NOT_FOUND: 404
};
var PlaceCardType;
(function (PlaceCardType) {
    PlaceCardType["DEFAULT"] = "DEFAULT";
    PlaceCardType["FAVORITE"] = "FAVORITE";
    PlaceCardType["NEARBY"] = "NEARBY";
})(PlaceCardType = exports.PlaceCardType || (exports.PlaceCardType = {}));
exports.EMPTY_REVIEW = {
    text: "",
    rating: 0,
    offerID: null
};
exports.BASE_URL = "https://htmlacademy-react-3.appspot.com/six-cities";
exports.TIME_OUT = 5000;
