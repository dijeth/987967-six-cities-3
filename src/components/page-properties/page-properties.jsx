import React from 'react';
import PropTypes from 'prop-types';
import {BREAK_STRING, MAX_IMAGE_COUNT} from '../../const/const';
import {reviewPropTypes} from '../../const/props.js';
import {ratingToPercent} from '../../util.js';
import ReviewList from '../review-list/review-list.jsx';
import ReviewForm from '../review-form/review-form.jsx';
import OffersNearby from '../offers-nearby/offers-nearby';
import OffersMap from '../offers-map/offers-map.jsx';
import {AppRoute} from '../../const/const';
import Header from '../header/header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {getComments, getNearbyCoordList} from '../../reducers/data/selectors.js';
import {Operation as DataOperation} from '../../reducers/data/operation.js';
import {getActiveOffer, getActiveOfferCoord} from '../../reducers/app/selectors.js';
import withLoading from '../../hocs/with-loading/with-loading.jsx';
import withPageError from '../../hocs/with-page-error/with-page-error.jsx';
import {Link, Redirect} from 'react-router-dom';
import Adapter from '../../adapter/adapter.js';

const PageProperties = ({offer, reviews, isAuthorized, activeCityCoord, offersCoord, onFavoriteChange}) => {
  if (offer === null) {
    return <Redirect to={AppRoute.getRoot()} />;
  }

  const {
    title,
    type,
    pictures,
    cost,
    rating,
    isPremium,
    isFavorite,
    bedroomCount,
    adultsCount,
    insideFeatures,
    userName,
    userPicture,
    isSuperUser,
    description,
    descriptionTitle,
    id,
    zoom
  } = offer;

  const gallery = pictures.slice(1, 1 + MAX_IMAGE_COUNT).map((it, i) => {
    return (
      <div className="property__image-wrapper" key={`${it}-${i}`}>
        <img className="property__image" src={it} alt="Photo studio" />
      </div>);
  });

  const ratingPercent = ratingToPercent(rating);

  const features = insideFeatures.map((it, i) => {
    return (
      <li className="property__inside-item" key={`${it}-${i}`}>
        {it}
      </li>);
  });

  const descriptionText = description.split(BREAK_STRING).map((it, i) => {
    return <p className="property__text" key={`${it}-${i}`}>{it}</p>;
  });

  const centerCoord = activeCityCoord;

  const favoriteButtonBlock = (
    <button
      className={`property__bookmark-button ${isFavorite ? `property__bookmark-button--active` : ``} button`}
      type="button"
      onClick={() => {
        onFavoriteChange(offer);
      }}
    >
      <svg className="property__bookmark-icon" width="31" height="33">
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </button>);

  const linkToLoginBlock = (
    <Link to={AppRoute.getLogin()} className={`property__bookmark-button button`} type="button">
      <svg className="property__bookmark-icon" width="31" height="33">
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </Link>);

  return (
    <div className="page">
      <Header isActiveLogo={false} />
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {gallery}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {isPremium && <div className="property__mark"><span>Premium</span></div>}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {title}
                </h1>
                {isAuthorized ? favoriteButtonBlock : linkToLoginBlock}
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${ratingPercent}%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {bedroomCount} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {adultsCount} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{cost}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {features}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">{descriptionTitle}</h2>
                <div className="property__host-user user">
                  <div className={`property__avatar-wrapper ${isSuperUser ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                    <img className="property__avatar user__avatar" src={`/${userPicture}`} width="74" height="74" alt="Host avatar" />
                  </div>
                  <span className="property__user-name">
                    {userName}
                  </span>
                </div>
                <div className="property__description">
                  {descriptionText}
                </div>
              </div>
              <section className="property__reviews reviews">
                {reviews && <ReviewList reviews={reviews} />}
                {isAuthorized && <ReviewForm offerID={id} />}
              </section>
            </div>
          </div>
          <section className="property__map map">
            <OffersMap centerCoord={centerCoord} offersCoord={offersCoord} zoom={zoom} />
          </section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              <OffersNearby nearPlacesFor={id} isAuth={isAuthorized} />
            </div>
          </section>
        </div>
      </main>
    </div>);
};

PageProperties.propTypes = {
  offer: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    pictures: PropTypes.arrayOf(PropTypes.string),
    cost: PropTypes.number.isRequired,
    rating: PropTypes.number,
    isPremium: PropTypes.bool,
    isFavorite: PropTypes.bool,
    city: PropTypes.string.isRequired,
    bedroomCount: PropTypes.number,
    adultsCount: PropTypes.number,
    insideFeatures: PropTypes.arrayOf(PropTypes.string),
    userName: PropTypes.string.isRequired,
    userPicture: PropTypes.string,
    isSuperUser: PropTypes.bool,
    descriptionTitle: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    coord: PropTypes.arrayOf(PropTypes.number),
    zoom: PropTypes.number
  }),

  reviews: PropTypes.arrayOf(PropTypes.shape(reviewPropTypes)),
  isAuthorized: PropTypes.bool.isRequired,
  activeCityCoord: PropTypes.arrayOf(PropTypes.number),
  onFavoriteChange: PropTypes.func,
  offersCoord: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number))
};

const mapStateToProps = (state) => ({
  offer: getActiveOffer(state),
  isAuthorized: getAuthorizationStatus(state),
  activeCityCoord: getActiveOfferCoord(state),
  reviews: getComments(state),
  offersCoord: getNearbyCoordList(state)
});

const mapDispatchToProps = (dispatch) => {
  return {
    onFavoriteChange(offer) {
      dispatch(DataOperation.changeFavorite(offer.id, Adapter.postFavorite(!offer.isFavorite)));
    }
  };
};

export {PageProperties};
export default withPageError(withLoading(connect(mapStateToProps, mapDispatchToProps)(PageProperties)));
