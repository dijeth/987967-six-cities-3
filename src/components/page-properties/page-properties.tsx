import * as React from 'react';
import {BREAK_STRING, MAX_IMAGE_COUNT} from '../../const/const';
import {ratingToPercent} from '../../util.js';
import ReviewList from '../review-list/review-list';
import ReviewForm from '../review-form/review-form';
import OffersNearby from '../offers-nearby/offers-nearby';
import OffersMap from '../offers-map/offers-map';
import {AppRoute} from '../../const/const';
import Header from '../header/header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {getComments, getNearbyCoordList} from '../../reducers/data/selectors.js';
import {Operation as DataOperation} from '../../reducers/data/operation.js';
import {getActiveOffer, getActiveOfferCoord} from '../../reducers/app/selectors.js';
import withLoading from '../../hocs/with-loading/with-loading';
import withPageError from '../../hocs/with-page-error/with-page-error';
import {Link, Redirect} from 'react-router-dom';
import Adapter from '../../adapter/adapter.js';
import {Offer, UserReview, OfferMini} from '../../interfaces';
import {coord} from '../../types';

type Props = {
  offer: Offer;
  reviews: Array<UserReview>;
  isAuthorized: boolean;
  activeCityCoord: coord;
  onFavoriteChange: (offer: OfferMini) => void;
  offersCoord: Array<coord>;
};

const PageProperties: React.FC<Props> = ({offer, reviews, isAuthorized, activeCityCoord, offersCoord, onFavoriteChange}) => {
  if (offer === null) {
    return <Redirect to={AppRoute.getRoot()} />;
  }

  const {
    title,
    type,
    pictures,
    cost,
    rating,
    isPremium,
    isFavorite,
    bedroomCount,
    adultsCount,
    insideFeatures,
    userName,
    userPicture,
    isSuperUser,
    description,
    descriptionTitle,
    id,
    zoom
  } = offer;

  const gallery = pictures.slice(1, 1 + MAX_IMAGE_COUNT).map((it, i) => {
    return (
      <div className="property__image-wrapper" key={`${it}-${i}`}>
        <img className="property__image" src={it} alt="Photo studio" />
      </div>);
  });

  const ratingPercent = ratingToPercent(rating);

  const features = insideFeatures.map((it, i) => {
    return (
      <li className="property__inside-item" key={`${it}-${i}`}>
        {it}
      </li>);
  });

  const descriptionText = description.split(BREAK_STRING).map((it, i) => {
    return <p className="property__text" key={`${it}-${i}`}>{it}</p>;
  });

  const centerCoord = activeCityCoord;

  const favoriteButtonBlock = (
    <button
      className={`property__bookmark-button ${isFavorite ? `property__bookmark-button--active` : ``} button`}
      type="button"
      onClick={() => {
        onFavoriteChange(offer);
      }}
    >
      <svg className="property__bookmark-icon" width="31" height="33">
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </button>);

  const linkToLoginBlock = (
    <Link to={AppRoute.getLogin()} className={`property__bookmark-button button`} type="button">
      <svg className="property__bookmark-icon" width="31" height="33">
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </Link>);

  return (
    <div className="page">
      <Header isActiveLogo={false} />
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {gallery}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {isPremium && <div className="property__mark"><span>Premium</span></div>}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {title}
                </h1>
                {isAuthorized ? favoriteButtonBlock : linkToLoginBlock}
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${ratingPercent}%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {bedroomCount} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {adultsCount} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{cost}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {features}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">{descriptionTitle}</h2>
                <div className="property__host-user user">
                  <div className={`property__avatar-wrapper ${isSuperUser ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                    <img className="property__avatar user__avatar" src={`/${userPicture}`} width="74" height="74" alt="Host avatar" />
                  </div>
                  <span className="property__user-name">
                    {userName}
                  </span>
                </div>
                <div className="property__description">
                  {descriptionText}
                </div>
              </div>
              <section className="property__reviews reviews">
                {reviews && <ReviewList reviews={reviews} />}
                {isAuthorized && <ReviewForm offerID={id} />}
              </section>
            </div>
          </div>
          <section className="property__map map">
            <OffersMap centerCoord={centerCoord} offersCoord={offersCoord} zoom={zoom} />
          </section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              <OffersNearby nearPlacesFor={id} isAuth={isAuthorized} />
            </div>
          </section>
        </div>
      </main>
    </div>);
};

const mapStateToProps = (state) => ({
  offer: getActiveOffer(state),
  isAuthorized: getAuthorizationStatus(state),
  activeCityCoord: getActiveOfferCoord(state),
  reviews: getComments(state),
  offersCoord: getNearbyCoordList(state)
});

const mapDispatchToProps = (dispatch) => {
  return {
    onFavoriteChange(offer) {
      dispatch(DataOperation.changeFavorite(offer.id, Adapter.postFavorite(!offer.isFavorite)));
    }
  };
};

export {PageProperties};
export default withPageError(withLoading(connect(mapStateToProps, mapDispatchToProps)(PageProperties)));
