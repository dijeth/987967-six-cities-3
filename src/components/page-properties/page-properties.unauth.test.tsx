import * as React from 'react';
import * as Enzyme from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';
import {PageProperties} from './page-properties';
import NameSpace from '../../reducers/name-space.js';
import {EMPTY_REVIEW} from '../../const/const';
import { Offer, OfferMini } from '../../interfaces.js';

const mockStore = configureStore([]);

const offers: Array<OfferMini> = [{
  city: `city-1`,
  pictures: [`picture-1`],
  title: `title-1`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 1,
  coord: [11, 111],
  id: `1`
},
{
  city: `city-1`,
  pictures: [`picture-2`],
  title: `title-2`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 2,
  coord: [22, 222],
  id: `2`
},
{
  city: `city-1`,
  pictures: [`picture-3`],
  title: `title-3`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 3,
  coord: [33, 333],
  id: `3`
},
{
  city: `city-1`,
  pictures: [`picture-4`],
  title: `title-4`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 4,
  coord: [44, 444],
  id: `4`
},
];

const neighbourhoods: Array<OfferMini> = [{
  city: `city-1`,
  pictures: [`picture-1`],
  title: `title-1`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 1,
  coord: [11, 111],
  id: `1`
},
{
  city: `city-1`,
  pictures: [`picture-2`],
  title: `title-2`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 2,
  coord: [22, 222],
  id: `2`
},
{
  city: `city-1`,
  pictures: [`picture-3`],
  title: `title-3`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `room`,
  cost: 3,
  coord: [33, 333],
  id: `3`
},
];

const activeOffer: Offer = {
  id: `id`,
  title: `title`,
  type: `room`,
  pictures: [`picture-1`, `picture-2`, `picture-3`, `picture-4`, `picture-5`, `picture-6`],
  cost: 123,
  rating: 1,
  isPremium: true,
  isFavorite: false,
  city: `city-1`,
  bedroomCount: 1,
  adultsCount: 1,
  insideFeatures: [`insideFeatures`],
  userName: `userName`,
  userPicture: `userPicture`,
  isSuperUser: true,
  descriptionTitle: `descriptionTitle`,
  description: `description`,
  coord: [1, 2],
  zoom: 1,
};

const store = mockStore({
  [NameSpace.USER]: {
    isAuthorized: false,
    userReviewText: EMPTY_REVIEW.text,
    userReviewRating: EMPTY_REVIEW.rating,
    userReviewOfferID: EMPTY_REVIEW.offerID,
  },
  [NameSpace.APP]: {
    activeOffer,
    activeCity: {
      name: `city-1`,
      centerCoord: [1, 2],
      zoom: 3
    }
  },
  [NameSpace.DATA]: {
    offers,
    nearbyList: neighbourhoods
  }
});

describe(`<PageProperties /> snapshot test`, () => {
  it(`should be rendered correctly with an unauthorized user`, () => {
    const div = document.createElement(`div`);
    document.body.appendChild(div);

    const tree = Enzyme.mount(
        <Provider store={store}>
          <BrowserRouter>
            <PageProperties
              isAuthorized={false}
              offer={activeOffer}
              activeCityCoord={[1, 2]}
              reviews={[]}
              offersCoord={neighbourhoods.map((it) => it.coord)}
              onFavoriteChange={()=>{}}
            />
          </BrowserRouter>
        </Provider>, {attachTo: div});

    expect(tree.getDOMNode()).toMatchSnapshot();
  });
});
