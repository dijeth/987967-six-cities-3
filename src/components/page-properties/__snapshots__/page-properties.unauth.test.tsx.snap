// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageProperties /> snapshot test should be rendered correctly with an unauthorized user 1`] = `
<div
  class="page"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper  user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--property"
  >
    <section
      class="property"
    >
      <div
        class="property__gallery-container container"
      >
        <div
          class="property__gallery"
        >
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="picture-2"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="picture-3"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="picture-4"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="picture-5"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="picture-6"
            />
          </div>
        </div>
      </div>
      <div
        class="property__container container"
      >
        <div
          class="property__wrapper"
        >
          <div
            class="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            class="property__name-wrapper"
          >
            <h1
              class="property__name"
            >
              title
            </h1>
            <a
              class="property__bookmark-button button"
              href="/login"
              type="button"
            >
              <svg
                class="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </a>
          </div>
          <div
            class="property__rating rating"
          >
            <div
              class="property__stars rating__stars"
            >
              <span
                style="width: 20%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              class="property__rating-value rating__value"
            >
              1
            </span>
          </div>
          <ul
            class="property__features"
          >
            <li
              class="property__feature property__feature--entire"
            >
              room
            </li>
            <li
              class="property__feature property__feature--bedrooms"
            >
              1
               Bedrooms
            </li>
            <li
              class="property__feature property__feature--adults"
            >
              Max 
              1
               adults
            </li>
          </ul>
          <div
            class="property__price"
          >
            <b
              class="property__price-value"
            >
              €
              123
            </b>
            <span
              class="property__price-text"
            >
               night
            </span>
          </div>
          <div
            class="property__inside"
          >
            <h2
              class="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              class="property__inside-list"
            >
              <li
                class="property__inside-item"
              >
                insideFeatures
              </li>
            </ul>
          </div>
          <div
            class="property__host"
          >
            <h2
              class="property__host-title"
            >
              descriptionTitle
            </h2>
            <div
              class="property__host-user user"
            >
              <div
                class="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  class="property__avatar user__avatar"
                  height="74"
                  src="/userPicture"
                  width="74"
                />
              </div>
              <span
                class="property__user-name"
              >
                userName
              </span>
            </div>
            <div
              class="property__description"
            >
              <p
                class="property__text"
              >
                description
              </p>
            </div>
          </div>
          <section
            class="property__reviews reviews"
          >
            <h2
              class="reviews__title"
            >
              Reviews · 
              <span
                class="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              class="reviews__list"
            />
          </section>
        </div>
      </div>
      <section
        class="property__map map"
      >
        <div
          class="leaflet-container leaflet-grab leaflet-touch-drag"
          id="map"
          style="width: 100%; height: 100%; position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 1;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://b.basemaps.cartocdn.com/rastertiles/voyager_labels_under/1/1/0{@2x}.png"
                    style="width: 256px; height: 256px; left: -3px; top: -255px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-shadow-pane"
            />
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            >
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="img/pin.svg"
                style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 155px; top: -15px; z-index: -15;"
                tabindex="0"
              />
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="img/pin.svg"
                style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 313px; top: -31px; z-index: -31;"
                tabindex="0"
              />
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="img/pin.svg"
                style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 471px; top: -49px; z-index: -49;"
                tabindex="0"
              />
            </div>
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            />
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JS library for interactive maps"
                >
                  Leaflet
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
    </section>
    <div
      class="container"
    >
      <section
        class="near-places places"
      >
        <h2
          class="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          class="near-places__list places__list"
        >
          <div
            class="near-places__list places__list"
          >
            <article
              class="near-places__card place-card"
            >
              <div
                class="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="picture-1"
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      1
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <a
                    class="place-card__bookmark-button button"
                    href="/login"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </a>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 20%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/1"
                  >
                    title-1
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
            <article
              class="near-places__card place-card"
            >
              <div
                class="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="picture-2"
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      2
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <a
                    class="place-card__bookmark-button button"
                    href="/login"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </a>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 20%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/2"
                  >
                    title-2
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
            <article
              class="near-places__card place-card"
            >
              <div
                class="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="picture-3"
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      3
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <a
                    class="place-card__bookmark-button button"
                    href="/login"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </a>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 20%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/3"
                  >
                    title-3
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  room
                </p>
              </div>
            </article>
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
